launcher/src/test/java/org/wso2/carbon/launcher/test/DropinsBundleDeployerTest.java

```diff
@@ -235,24 +236,17 @@ private static boolean compareBundleInfo(List<BundleInfo> expected, List<BundleI
                         .count()) == expected.size());
     }
 
-    private static boolean delete(Path path) throws IOException {
-        if (Files.isDirectory(path)) {
-            List<Path> children = listFiles(path);
-            if (children.size() > 0) {
-                for (Path aChild : children) {
-                    delete(aChild);
-                }
-            }
-        }
-        return Files.deleteIfExists(path);
+    private static void delete(Path path) throws IOException {
+        Path osgiRepoPath = Paths.get(carbonHome, Constants.OSGI_REPOSITORY);
+        Files.list(path)
+                .filter(child -> !osgiRepoPath.equals(child) && Files.isDirectory(child))
+                .forEach(child -> {
+                    try {
+                        FileUtils.deleteDirectory(child.toFile());
+                    } catch (IOException e) {
+                        throw new UncheckedIOException(e);
```

Comment:
I couldn't find any other way to handle the IOException thrown by deleteDirectory() with java8 lambda expression other than [1]. Could you please suggest if there is any better way.

[1. ](http://stackoverflow.com/questions/19757300/java-8-lambda-streams-filter-by-method-with-exception)

Thanks.